### PR TITLE
[Model] use FusedMoE layer in Jamba

### PR DESCRIPTION
**Pull request by avshalomman**
This PR replaces the Jamba-specific MoE layer impl with the standard vLLM FusedMoE layer.

### CODE REVIEW

**Comment by robertgshaw2-neuralmagic** [comment]
Thanks for making this change! Did not realize Jamba had `FusedMoE` during the refactor

**Comment by avshalomman** [comment]
I see that some tests in the extended suite are failing but they don't seem to be related

File: vllm/model_executor/models/jamba.py [hunk 1]
```diff
@@ -917,15 +858,13 @@ def load_weights(self, weights: Iterable[Tuple[str, torch.Tensor]]):
...
-        expert_params_mapping = [
-            # (param_name, weight_name, expert_id)
-            (
-                "ws" if weight_name in ["gate_proj", "up_proj"] else "w2s",
-                f"experts.{expert_id}.{weight_name}.weight",
-                expert_id,
-            ) for expert_id in range(self.config.num_experts)
-            for weight_name in ["down_proj", "up_proj", "gate_proj"]
-        ]
+        # Params for weights, fp8 weight scales, fp8 activation scales
```
Comment (mgoin) [inline vllm/model_executor/models/jamba.py:861]: Nit: Have you tested FP8 quantization with this model?

**Review by mgoin — COMMENTED** [review]
LGTM but is there an active test in CI to ensure this works?

File: vllm/model_executor/models/jamba.py (hunk 1 above)
Comment (avshalomman) [reply vllm/model_executor/models/jamba.py:861]: Hey yes we have, internally

**Comment by avshalomman** [comment]
> LGTM but is there an active test in CI to ensure this works?

There are 10 general tests for the Jamba model, which pass after the change to the FusedMoE layer

**Review by mgoin — APPROVED** [review]
Thanks for the answers, LGTM

**Comment by avshalomman** [comment]
> Thanks for the answers, LGTM

Thanks! How can I re-run the failing tests? These seem like transient, unrelated errors

**Pull request merged**